share/revng/pipelines/revng-pipelines.yml

```diff
@@ -660,6 +660,61 @@ Branches:
           SingleTargetFilename: mlir-llvm-dialect-with-clift-types.mlir
           Docs: |-
             A test artifact to import types into a Clift module.
+    # New clift-based decompilation pipeline
+  - From: detect-stack-size
+    Steps:
+      - Name: segregate-stack-accesses
+        Pipes:
+          - Type: llvm-pipe
+            UsedContainers: [module.bc.zstd]
+            Passes:
+              - hoist-struct-phis
+              - segregate-stack-accesses
+      - Name: late-optimize
+        Pipes:
+          - Type: llvm-pipe
+            UsedContainers: [module.bc.zstd]
+            Passes:
+              - cleanup-stack-size-markers
+              - dce
+              - sroa
+              - instcombine
+              - remove-extractvalues
+              - sroa
+              - simplify-cfg-with-hoist-and-sink
+              # loop-rotate rearranges loops in a way that should make it
+              # easier to match do-while loops
+              - loop-rotate
+              - loop-rewrite-with-canonical-induction-variable
+              - simplify-cfg-with-hoist-and-sink
+              # don't run simplify-cfg{,-with-hoist-and-sink} after
+              # loop-simplify because it kills the loop-simplify form causing
+              # DLA not to identify arrays properly
+              - loop-simplify
+              - instcombine
+              - remove-extractvalues
+              - early-cse
+              - dce
+              - strip-dead-prototypes
+              - split-overflow-intrinsics
+              - dce
+      - Name: switch-to-statements
+        Pipes:
+          - Type: llvm-pipe
+            UsedContainers: [module.bc.zstd]
+            Passes:
+              - remove-llvmassume-calls
+              - dce
+              - exit-ssa
+              - switch-to-statements
+        Artifacts:
+          Container: module.bc.zstd
+          Kind: stack-accesses-segregated
+          SingleTargetFilename: stack_access_segregate.ll
+          Docs: |-
+            This artifact contains an LLVM function for each function in the
+            input program.
+            TODO: fill this up
```

Comment:
Do not leave TODOs for documentation, this is user facing!

Reply:
Done!